internal/livelint/run_checks.go

```diff
@@ -302,6 +307,24 @@ func (n *Livelint) RunChecks(namespace, deploymentName string, isVerbose bool) e
 			return nil
 		}
 
+		result = checkHasValidIngressClass(ingress, ingressClasses)
+		statusMsg.AddCheckResult(result)
+		n.ui.Send(statusMsg)
+		if result.HasFailed {
+			return nil
+		}
+
+		result, err = n.checkCanAccessAppFromIngressController(ingress, ingressClasses)
```

Comment:
Why do we diverge from the flow chart here?

Reply:
There are two points where we diverge:

1. additional check `checkHasValidIngressClass`: this is an opportunity to report an additional potential error. This only comes up when you are trying to determine the ingress controller pods which are responsible for a particular `Ingress` resource via the API.  I have adapted the flow chart for this.
2. failure branch after `checkCanSeeBackends`: this is simply not implemented yet. I have opened #71 for this.